### PR TITLE
No monkeypatched helper functions on collections

### PR DESCRIPTION
**Pull request by jimrandomh**
Problem: Lots of functions defined by adding them to collections, rather than as normal functions imported in the normal way. This will cause problems as we move away from Meteor and as we try to limit the scope of which code has to run on the client.

Solution: Turn them into regular function calls. So `Posts.getUrl` becomes `postGetUrl`.

Not in scope of this PR: Things monkeypatched onto specific collections which are not functions (eg, post statuses); `toAlgolia` (which is accessed via a variable collection); `Utils` (which does a similar bad thing, but is not a collection).

### CODE REVIEW

File: packages/lesswrong/server/ckEditorToken.ts [hunk 1]
```diff
@@ -20,7 +21,7 @@ addStaticRoute('/ckeditor-token', async ({ query }, req, res, next) => {
 
   const user = await getUserFromReq(req)
   const post = await Posts.findOne(documentId)
-  const canEdit = post && Posts.canEdit(user, post)  
+  const canEdit = post && postCanEdit(user, post)
```
Comment (Discordius) [inline packages/lesswrong/server/ckEditorToken.ts:24]: Aren't these `canEdit` `canRead` and `canDelete` function used in a bunch of internal Collection specific Vulcan code?

Comment (Discordius) [reply packages/lesswrong/server/ckEditorToken.ts:24]: Nope, I am wrong, apparently it isn't

Comment (jimrandomh) [reply packages/lesswrong/server/ckEditorToken.ts:24]: I did a search on `canEdit`, and it doesn't have any accesses like that (there are a few variables that pun it, but no members or member accesses). `checkAccess`, on the other hand, is vulcan-internal in the way you're thinking, as is very-similarly-named `canUpdate`.

Comment (Discordius) [reply packages/lesswrong/server/ckEditorToken.ts:24]: Ah, cool. I was probably thinking of `canUpdate`

**Comment by jimrandomh** [comment]
Fixed a few references that escaped the type system because they were written as `getCollection(...).functionName` instead of `Collection.functionName`, found by searching for all usages of any of the moved identifiers.

Also note, I haven't done `Users` yet, which has the trickiest ones (since its functions can potentially be both Vulcan-special and also Meteor-special).

**Comment by jimrandomh** [comment]
Now running into a unit test failure caused by cyclic imports (introduced by changing what imports what). Going to turn on the lint rule and break some of those cycles.

**Comment by Discordius** [comment]
I guess I reviewed this as well as I am going to? It's reasonably high risk, because of how much code is touched, and because it changes various import order stuff. My guess is the only way to really merge this one is to just do a bunch of testing.

**Review by Discordius — APPROVED** [review]

**Pull request merged**